### PR TITLE
H-586, H-673: Style cleanup in top-nav dropdowns

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

Addressing H-586 and H-673, this PR makes some improvements to the navbar dropdowns

## 🔍 What does this change?

- moves the "Settings" link to the account dropdown
- adds a "Create link type" button to the action dropdown
- updates the copy in the new entity type page to reflect when a link type is being created
- adds an icon in the action dropdown for the "Create entity" button

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## 🛡 What tests cover this?

None

## ❓ How to test this?

View the deployment and test out the new dropdown options.

## 📹 Demo

image
image

### CODE REVIEW

File: apps/hash-frontend/src/pages/new/types/entity-type.page.tsx [hunk 1]
```diff
@@ -41,7 +45,7 @@ const Page: NextPageWithLayout = () => {
               id: "types",
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/new/types/entity-type.page.tsx:45]:
```suggestion
              id: "types",
              href: "/types",
```

File: apps/hash-frontend/src/pages/new/types/entity-type.page.tsx [hunk 2]
```diff
@@ -41,7 +45,7 @@ const Page: NextPageWithLayout = () => {
...
             },
             {
-              title: "Entity types",
+              title: `${isCreateLinkEntityType ? "Link" : "Entity"} types`,
               href: "#",
```
Comment (CiaranMn) [inline apps/hash-frontend/src/pages/new/types/entity-type.page.tsx:49]:
```suggestion
              href: "/types/link-type",
```

**Review by CiaranMn — COMMENTED** [review]
A couple of suggestions for breadcrumbs.

On the appearance, @nonparibus will comment, but I think:
1. we should make it `Create page` for casing consistency
2. Looks a bit weird to have a single icon, would have icons for all or none. If all, the questions are then:
   - whether the icons should be aligned to the right (in which case there will be a large gap between entity/page and the icon)
   - what icons for the others, to which I believe the answers are:
      - link type: `LinkIcon`
      - entity type: `AsteriskIcon` (there's also an `EntityTypeIcon` in the design system)
      - page type: `PageIcon`

**Edit**
I've found a [design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH?type=design&node-id=8940-74543&mode=design&t=0Ivrac8pMQDNBww3-4) for this that suggests there are left-aligned icons and different dividing lines to how it is at the moment.

Screenshot 2023-09-14 at 10 36 47

The 'Upload file' link (there's a separate task I created for upload file in the header nav, H-673, but could be done as part of this) should go to `${types.entityType.file.entityTypeId}?tab=upload`. We could potentially have an additional one for image uploads, which would be `${types.entityType.image.entityTypeId}?tab=upload`

**Comment by benwerner01** [comment]
> I've found a [design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH?type=design&node-id=8940-74543&mode=design&t=0Ivrac8pMQDNBww3-4) for this that suggests there are left-aligned icons and different dividing lines to how it is at the moment.

@CiaranMn ah thanks, I wasn't aware of those. The dividing lines in the designs conflict with the instructions from the ticket, but I'll reconcile those differences and will leave to @nonparibus to decide when he does a final review

**Comment by nonparibus** [comment]
Can we add the “Sync external data” link, too, and have it point to the “add integration” page?

Until we have the unified “create type” page, can we replace the “link type” icon mocked above with the `link` icon we use?

**Comment by CiaranMn** [comment]
@nonparibus 
> Can we add the “Sync external data” link, too, and have it point to the “add integration” page?

Can we defer this until it is functional / ready? We don't want to anyone to actually use this at the moment since there is no authorization. 

> Let’s not add an additional upload link for images specifically. Instead, when a user uploads a file we should auto-infer the correct additional other child type(s) to be applied.
I've created [H-780](https://linear.app/hash/issue/H-780/create-an-image-instead-of-a-file-type-if-a-user-uploads-an-image) for this

**Comment by nonparibus** [comment]
Fine from my side ^

**Comment by benwerner01** [comment]
> Until we have the unified “create type” page, can we replace the “link type” icon mocked above with the link icon we use?

@nonparibus sure, I'm assuming you mean this icon? Added in https://github.com/hashintel/hash/pull/3150/commits/41538a25ad06492f415b7184bcd16baaa5de6724

image

> Can we also update the default/hover/active/etc states of the currently blue +/avatar icons in the header navbar itself, to match the Figma?

Sure, have done so in https://github.com/hashintel/hash/pull/3150/commits/45f0b4267da68cee7e04da5c6a9e49512c4d0336. Now it looks like this:

https://github.com/hashintel/hash/assets/42802102/2cf9939f-79f8-4b47-9817-fc1a70ecf3b3

**Review by CiaranMn — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**